pkg/metaclient/example.go

```diff
@@ -0,0 +1,174 @@
+	// expect len(getRsp.Kvs) == 1
+	// expect getRsp.Kvs[0].TTL == 5
+	// sleep 3s
+	getRsp, err = cli.Get(ctx, "data")
+	// expect len(getRsp.Kvs) == 1
+	// expect getRsp.Kvs[0].TTL == 2
+	// sleep 3s
+	getRsp, err = cli.Get(ctx, "data")
+	// expect len(getRsp.Kvs) == 0, key timeout
+
+	// Key Range, forbit Put using WithRange
+	// current data:
+	//		apple  red
+	//		orange orange
+	//		ticdc  kv
+	//		dm	   DDL
+	getRsp, err = cli.Get(ctx, "ap", WithRange("zz"))
+	// expect len(getRsp.Kvs) == 4(apple. orange, ticdc, dm)
+	getRsp, err = cli.Get(ctx, "ap", WithRange("apple2"))
+	// expect len(getRsp.Kvs) == 1(apple)
+	delRsp, err = cli.Delete(ctx, "dzst", WithRange("panda"))
+	// delete key orange
+
+	// From Key, forbit Put using WithFromKey
+	// current data:
+	//		apple  red
+	//		ticdc  kv
+	//		dm	   DDL
+	getRsp, err = cli.Get(ctx, "data", WithFromKey())
+	// expect len(getRsp.Kvs) == 2(ticdc, dm)
+	delRsp, err = cli.Delete(ctx, "tian", WithFromKey())
+	// delete key ticdc
+
+	// Key Prefix, forbit Put using WithPrefix
+	// current data:
+	//		apple  red
+	//		apple2  green
+	//		ticdc  kv
+	//		dm	   DDL
+	getRsp, err = cli.Get(ctx, "apple", WithPrefix())
+	// expect len(getRsp.Kvs) == 2(apple, apple2)
+	delRsp, err = cli.Delete(ctx, "apple", WithPrefix())
+	// delete key apple, apple2
+
+	// Sort, forbit Put/Delete using WithSort
+	// current data:
+	//		apple  red
+	//		apple2  green
+	//		ticdc  kv
+	//		dm	   DDL
+	getRsp, err = cli.Get(ctx, "apple", WithRange("zz"), WithSort(SortByKey, SortDescend))
+	// expect len(getRsp.Kvs) == 4(ticdc, dm, apple2, apple)
+	getRsp, err = cli.Get(ctx, "apple", WithRange("zz"), WithSort(SortByKey, SortAscend))
+	// expect len(getRsp.Kvs) == 4(apple, apple2, dm, ticdc)
+
+	// Limit, forbit Put/Delete using WithLimit
+	// current data:
+	//		apple  red
+	//		apple2  green
+	//		ticdc  kv
+	//		dm	   DDL
+	getRsp, err = cli.Get(ctx, "appleyy", WithRange("zz"), WithSort(SortByKey, SortDescend), WithLimit(2))
+	// expect len(getRsp.Kvs) == 2(ticdc, dm)
+	getRsp, err = cli.Get(ctx, "dm2", WithRange("zz"), WithSort(SortByKey, SortAscend), WithLimit(1))
+	// expect len(getRsp.Kvs) == 1(ticdc)
+
+	//
+	// Basic Do
+	//
+	// current data:
+	//		apple  red
+	//		apple2  green
+	//		ticdc  kv
+	//		dm	   DDL
+	getOp, err := OpGet("apple3", WithRange("zz"), WithSort(SortByKey, SortDescend), WithLimit(2))
+	opRsp, err := cli.Do(ctx, getOp)
+	getRsp = opRsp.Get()
+	// expect len(getRsp.Kvs) == 2(ticdc, dm)
+	putOp, err := OpPut("apple3", "t3", WithTTL(30))
+	opRsp, err = cli.Do(ctx, putOp)
+	putRsp = opRsp.Put()
+	// expect put: apple3 | t3| 30| xxx|
+	delOp, err := OpDelete("apple3", WithRange("ti"))
+	opRsp, err = cli.Do(ctx, delOp)
+	delRsp = opRsp.Del()
+	// expect delete: dm
+
+	//
+	// Txn, forbit nested txn
+	//
+	// current data:
+	//		apple  red
+	//		apple2  green
+	//		ticdc  kv
+	//		dm	   DDL
+	txn := cli.Txn(ctx)
+	txnRsp, err := txn.Do(getOp).Do(putOp).Do(delOp).Commit()
```

Comment:
Current `Txn` doesn't support the following scenarios
- etcd like compare, such as `Txn(...).If(...).Then(...).Commit()
- SQL like where condition in op

Reply:
Given the requirement from DM, https://github.com/pingcap/tiflow/blob/85116e4e6c973c8f82164908bc1041ec56d2fb1a/dm/pkg/shardddl/optimism/operation.go#L121-L123

Supposing the backend is a MySQL like database, we can provide a txn interface that allows user to write raw SQL, or supporting where condition in put op.

In this way we can keep the lightweight of kv interface, which provides simple API such [TiKV RawKV API](https://tikv.org/docs/5.1/develop/rawkv/introduction/), and a customized txn API.